src/scripts/main.js

```diff
@@ -1,3 +1,246 @@
+
+  let found = false;
+
+  while (!found) {
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (field[r][c] === 0) {
+      field[r][c] = 2;
+      renderField();
+
+      found = true;
+    }
+  }
+};
+
+function filterZero(row) {
+  return row.filter(num => num !== 0);
+};
+
+function moved(row) {
+  let newRow = row;
+
+  newRow = filterZero(newRow);
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+      score += newRow[i];
+    }
+  }
+
+  newRow = filterZero(newRow);
+
+  while (newRow.length < columns) {
+    newRow.push(0);
+  }
+
+  return newRow;
+};
+
+function movedLeft() {
+  for (let r = 0; r < rows; r++) {
+    let row = field[r];
+
+    row = moved(row);
+    field[r] = row;
+
+    renderField();
+  };
+};
+
+function movedRight() {
+  for (let r = 0; r < rows; r++) {
+    let row = field[r];
+
+    row.reverse();
+
+    row = moved(row);
+
+    row.reverse();
+    field[r] = row;
+
+    renderField();
+  };
+};
+
+function movedUp() {
+  for (let c = 0; c < columns; c++) {
+    let row = [field[0][c], field[1][c], field[2][c], field[3][c]];
+
+    row = moved(row);
+
+    for (let r = 0; r < rows; r++) {
+      field[r][c] = row[r];
+    }
+    renderField();
+  };
+};
+
+function movedDown() {
+  for (let c = 0; c < columns; c++) {
+    let row = [field[0][c], field[1][c], field[2][c], field[3][c]];
+
+    row.reverse();
+    row = moved(row);
+    row.reverse();
+
+    for (let r = 0; r < rows; r++) {
+      field[r][c] = row[r];
+    }
+    renderField();
+  };
+};
+
+function renderField() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      table.rows[r].cells[c].className = '';
+      table.rows[r].cells[c].classList.add('field-cell');
+      table.rows[r].cells[c].classList.add(`field-cell--${field[r][c]}`);
+      table.rows[r].cells[c].textContent = field[r][c] || '';
+    }
+  }
+};
+
+function checkWin() {
+  for (let i = 0; i < field[0].length; i++) {
+    for (let j = 0; j < field.length; j++) {
+      if (field[i][j] === 2048) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function isGameOver(board) {
+  const size = board.length;
+
+  for (let row = 0; row < size; row++) {
+    for (let col = 0; col < size; col++) {
+      if (board[row][col] === 0) {
+        return false;
+      }
+    }
+  }
+
+  for (let row = 0; row < size; row++) {
+    for (let col = 0; col < size; col++) {
+      const currentCell = board[row][col];
+
+      if (row > 0 && board[row - 1][col] === currentCell) {
+        return false;
+      }
+
+      if (row < size - 1 && board[row + 1][col] === currentCell) {
+        return false;
+      }
+
+      if (col > 0 && board[row][col - 1] === currentCell) {
+        return false;
+      }
+
+      if (col < size - 1 && board[row][col + 1] === currentCell) {
+        return false;
+      }
```

Comment:
Combine similar cases into 1, if condition will be so long use helper variables with meaningful variable naming 
![image](https://github.com/mate-academy/js_2048_game/assets/70009205/d6d5aedb-28ee-4b53-b049-2170d41e7968)